adapters/richaudience/richaudience.go

```diff
@@ -0,0 +1,355 @@
+			}
+
+			if raiExt.BidFloor <= 0 {
+				imp.BidFloor = 0.00001
+			} else {
+				imp.BidFloor = raiExt.BidFloor
+			}
+
+			if raiExt.BidFloorCur == "" {
+				imp.BidFloorCur = "USD"
+			} else {
+				imp.BidFloorCur = raiExt.BidFloorCur
+			}
+		}
+
+		if request.Site != nil && request.Site.Page != "" {
+			pageURL, error := url.Parse(request.Site.Page)
+			if error == nil && pageURL.Scheme == "https" {
+				secure = int8(1)
+			}
+		}
+		imp.Secure = &secure
+
+		if imp.Banner == nil {
+			err = &errortypes.BadInput{
+				Message: "Banner Object not found",
+			}
+			return nil, err
+		} else {
+			raiBanner := *imp.Banner
+			if raiBanner.W == nil && raiBanner.H == nil {
+				if len(raiBanner.Format) == 0 {
+					err = &errortypes.BadInput{
+						Message: "Format Object not found",
+					}
+					return nil, err
+				} else {
+					imp.Banner = &raiBanner
+				}
+			}
+		}
+
+		resImps = append(resImps, imp)
+
+	}
+	return resImps, nil
+}
+
+func setSite(request *openrtb2.BidRequest, richaudienceRequest *richaudienceRequest) {
+	if request.Site != nil {
+		if request.Site.Page != "" {
+			if request.Site.Domain == "" {
+				pageURL, error := url.Parse(request.Site.Page)
+				if error == nil {
+					richaudienceRequest.Site.Domain = pageURL.Host
+				}
+			} else {
+				richaudienceRequest.Site.Domain = request.Site.Domain
+			}
+
+			richaudienceRequest.Site.Page = request.Site.Page
+		}
+	}
+}
+
+func setDevice(request *openrtb2.BidRequest, richaudienceRequest *richaudienceRequest) (err error) {
+
+	if request.Device.IP == "" {
+		err = &errortypes.BadInput{
+			Message: "request.Device.IP is required",
+		}
+		return err
+	} else {
+		richaudienceRequest.Device.IP = request.Device.IP
+	}
+
+	if request.Device.DNT != nil {
+		richaudienceRequest.Device.DNT = *request.Device.DNT
+	} else {
+		richaudienceRequest.Device.DNT = 0
+	}
+
+	if request.Device.Lmt != nil {
+		richaudienceRequest.Device.Lmt = *request.Device.Lmt
+	} else {
+		richaudienceRequest.Device.Lmt = 0
+	}
+
+	if request.Device.UA != "" {
+		richaudienceRequest.Device.UA = request.Device.UA
+	}
+
+	return err
+}
+
+func setUser(request *openrtb2.BidRequest, richaudienceRequest *richaudienceRequest, requestInfo *adapters.ExtraRequestInfo) (err error) {
+	if request.User != nil {
+		if requestInfo.PbsEntryPoint != metrics.ReqTypeAMP {
+			if request.User.BuyerUID != "" && request.User.BuyerUID != "[PDID]" {
+				richaudienceRequest.User.BuyerUID = request.User.BuyerUID
+			} else {
+				err = &errortypes.BadInput{
+					Message: "request.ext.user.buyerUID is required",
+				}
+			}
+		}
+
+		if request.User.Ext != nil {
+			var extUser openrtb_ext.ExtUser
+			if err := json.Unmarshal(request.User.Ext, &extUser); err != nil {
+				return err
+			} else {
+				if extUser.Eids != nil {
+					richaudienceRequest.User.Ext.Eids = extUser.Eids
+				}
+				if extUser.Consent != "" {
+					richaudienceRequest.User.Ext.Consent = extUser.Consent
+				}
+			}
+		}
+	}
+
+	return
```

Comment:
For clarity, please change this to `return nil`.

Reply:
in next commit i have solved